src/main/java/org/opensearch/security/action/apitokens/ApiTokenRepository.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.apitokens;
+
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+
+public class ApiTokenRepository {
+    private final ApiTokenIndexHandler apiTokenIndexHandler;
+
+    public ApiTokenRepository(Client client, ClusterService clusterService) {
+        apiTokenIndexHandler = new ApiTokenIndexHandler(client, clusterService);
+    }
+
+    public String createApiToken(String name, List<String> clusterPermissions, List<ApiToken.IndexPermission> indexPermissions) {
+        apiTokenIndexHandler.createApiTokenIndexIfAbsent();
```

Comment:
i think we should create this index on bootstrap. Can be done in later PRs.

Reply:
Sure I was thinking about that - I will add a TODO here to circle back to make a decision.